src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -1492,3 +1492,35 @@ const Gaffer::GraphComponent *GafferScene::EditScopeAlgo::renderPassesReadOnlyRe
 
 	return MetadataAlgo::readOnlyReason( scope );
 }
+
+void GafferScene::EditScopeAlgo::renameRenderPass( Gaffer::EditScope *scope, const std::string &oldName, const std::string &newName )
+{
+	auto renderPassesProcessor = scope->acquireProcessor( g_renderPassesProcessorName, /* createIfNecessary = */ false );
+	if( !renderPassesProcessor )
+	{
+		return;
+	}
+
+	auto namesPlug = renderPassesProcessor->getChild<StringVectorDataPlug>( "names" );
+	ConstStringVectorDataPtr renderPasses = namesPlug->getValue();
+	if( std::find( renderPasses->readable().begin(), renderPasses->readable().end(), newName ) != renderPasses->readable().end() )
+	{
+		throw IECore::Exception( fmt::format( "Render Pass \"{}\" already exists", newName ) );
+	}
+
+	if( std::find( renderPasses->readable().begin(), renderPasses->readable().end(), oldName ) != renderPasses->readable().end() )
+	{
+		auto renderPassesCopy = renderPasses->copy();
+		std::replace( renderPassesCopy->writable().begin(), renderPassesCopy->writable().end(), oldName, newName );
+		namesPlug->setValue( renderPassesCopy );
+	}
```

Comment:
It seems a little odd that we're willing to ignore the absence of `oldName` here, but not to ignore the absence of `renderPassesProcessor`. Both cases are semantically the same - they mean the pass wasn't created here.

> I did wonder if `renameRenderPass()` should also be able to rename RenderPassOptionEdits in an EditScope without a RenderPasses processor

I think I like that - it would resolve the ambiguity above, and would allow someone to attempt the mind-reading "fix up my whole graph for me" operation.

Reply:
> I think I like that - it would resolve the ambiguity above, and would allow someone to attempt the mind-reading "fix up my whole graph for me" operation.

Cool, someone attempting that style of fix ups was what came to mind there. I've addressed some of the ambiguity, supported renaming RenderPassOptionEdits without a RenderPasses processor and added the return true/false suggestion below in 8edae92.

Last bit of ambiguity is whether we should also be throwing if there's a collision with `newName` already existing in RenderPassOptionEdits...

Reply:
> Last bit of ambiguity is whether we should also be throwing if there's a collision with newName already existing in RenderPassOptionEdits...

Oof. I think I would say that a single-row-per-pass is a guarantee we want the API to provide, so we should be throwing. We'd need to check before updating the RenderPasses node as well, so there aren't any side-effects when we do throw.

But what does that mean for the UI?

Reply:
Updated to throw when `newName` already exists in RenderPassOptionEdits and added `renameRenderPassNonEditableReason` (bit of a mouthful there) to return the reason in f88d94e.

> But what does that mean for the UI?

d2b2360 makes the UI display the reason as a popup after you try and rename to a colliding name. I think it would be better to prevent the user from clicking `Rename` in the dialogue like what happens when they enter a render pass name that collides with the names in the `renderPass:names` option, but the post-popup was a bit more straightforward than attempting to stick an additional check into a dialogue that's being shared with the add render pass functionality...